service-front/module/Application/src/Controller/IndexController.php

```diff
@@ -82,24 +91,37 @@ public function startAction(): Response
         return $this->redirect()->toRoute($redirect, ['uuid' => $case['uuid']]);
     }
 
+    private function constructDetailsDataBeforeCreatedCase(array $lpa, string $personType): array
+    {
+        $processed = $this->siriusDataProcessorHelper->processLpaResponse(
+            $personType,
+            $lpa
+        );
+
+        return [
+            'personType' => $personType,
+            'firstName' => $processed['first_name'],
+            'lastName' => $processed['last_name'],
+        ];
+    }
+
     /**
      * @param string $type
      * @psalm-param Lpa $lpaData
-     * @return void
+     * @return string
      * @throws HttpException
      */
-    private function checkStatusOfLpaIsStartable(string $type, array $lpaData): void
+    private function checkStatusOfLpaIsStartable(string $type, array $lpaData): string
     {
         try {
             $lpaStatusCheck = new LpaStatusTypeHelper($lpaData, $type);
 
             if (! $lpaStatusCheck->isStartable()) {
-                $errorMessage = $lpaStatusCheck->getStatus() === 'registered' ?
-                    "ID check has already been completed" :
+                return $lpaStatusCheck->getStatus() === 'registered' ?
+                    "The identity check has already been completed" :
                     "ID check has status: " . $lpaStatusCheck->getStatus() . " and cannot be started";
-
-                throw new HttpException(400, $errorMessage);
             }
+            return 'ok';
         } catch (\Exception $exception) {
             throw new HttpException(400, $exception->getMessage());
         }
```

Comment:
This method is a blending a bit too much together (whether it's startable, and what the error message should be) and it's only called in one place. I'd suggesting splitting it into two, but then you'd need to generate `$lpaStatusCheck` twice (once to return `->isStartable()`, once to calculate the message). Could the logic just be merged into `startAction`?